test/spec/Feature/Query/RpcSpec.hs

```diff
@@ -1447,3 +1447,15 @@ spec actualPgVersion =
               { matchStatus  = 400
               , matchHeaders = [matchContentTypeJson]
               }
+
+    -- https://github.com/PostgREST/postgrest/issues/1586#issuecomment-696345442
+    context "a proc with bit and char parameters" $ do
+      it "modifies the param type from character to character varying" $ do
+        get "/rpc/char_param?char_=abcdefg&char_arr={abc,abcdefg}" `shouldRespondWith`
+          [json| [{ "char_": "abcdefg", "char_arr": [ "abc", "abcdefg" ] }] |]
+          { matchHeaders = [matchContentTypeJson] }
+
+      it "modifies the param type from bit to bit varying" $ do
```

Comment:
How about a test that proves failure if the char exceeds the size?

Reply:
Added the test for the use case mentioned in the issue.